### PR TITLE
Implement feature request to support UseDefaultCredentials (54658)

### PR DESCRIPTION
**Pull request by ShachafGoldstein**
##### SUMMARY
Implement feature request to support UseDefaultCredentials as mentioned in issue https://github.com/ansible/ansible/issues/54658

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
win_uri

### CODE REVIEW

File: lib/ansible/modules/windows/win_uri.py [hunk 1]
```diff
@@ -148,6 +148,12 @@
       secure web request.
     type: str
     version_added: '2.5'
+  use_default_credentials:
+    description:
+    - Set the the ability of UseDefaultCredentials.
```
Comment (jhawkesworth) [inline lib/ansible/modules/windows/win_uri.py]: typo 'the the'.  Suggest 'Use default windows credentials for authentication.  Note that you must be using kerberos, ntlm or credssp for this to take effect.

(see https://docs.microsoft.com/en-us/dotnet/api/system.net.credentialcache.defaultcredentials?view=netframework-4.8#remarks)

**Comment by jhawkesworth** [comment]
This looks like a nice addition.
Consider whether this should be added to `win_get_url` as wel (in a separate PR).  If so it should have the same parameter name.
Having read https://docs.microsoft.com/en-us/dotnet/api/system.net.credentialcache.defaultcredentials?view=netframework-4.8#remarks I am not sure what the default credential would be as I think the login type will not be interactive unless you use 'become'.  This may not be an issue as I don't fully understand windows login authentication mechanism, but I thought it was worth asking.

File: lib/ansible/modules/windows/win_uri.py (hunk 1 above)
Comment (ShachafGoldstein) [reply lib/ansible/modules/windows/win_uri.py]: changed

**Comment by ShachafGoldstein** [comment]
> This looks like a nice addition.
> Consider whether this should be added to `win_get_url` as wel (in a separate PR). If so it should have the same parameter name.
> Having read https://docs.microsoft.com/en-us/dotnet/api/system.net.credentialcache.defaultcredentials?view=netframework-4.8#remarks I am not sure what the default credential would be as I think the login type will not be interactive unless you use 'become'. This may not be an issue as I don't fully understand windows login authentication mechanism, but I thought it was worth asking.

Since there is the issue of double hop in Kerberos you'll have to use become, for NTLM or CREDSSP (when configured correctly) should work.

Maybe we should consider merging the two singe httpwebrequest is derived from web request 
(win_uri)

**Comment by jhawkesworth** [comment]
Please consider adding an example that uses 'become' to demonstrate usage, if 'become' is actually required.

**Comment by ShachafGoldstein** [comment]
Should test KERB, NTLM, CREDSSP and update docs accordingly

**Comment by jborean93** [comment]
@ShachafGoldstein are you ok with adding this functionality with https://github.com/ansible/ansible/pull/54759 instead? It adds this particular property as well as a few others to both win_get_url and win_uri.

**Comment by ShachafGoldstein** [comment]
Yeah sure, I never finished testing all cred scenarios though

**Comment by jborean93** [comment]
Cool will close this one then, if you get a change it would be great if you could review/test out #54759

**Pull request closed without merging**